### DISCUSSION TITLE
Bulk Action Select All, so slow on large data

### DISCUSSION THREAD

**Discussion by frascthx** [rappasoft/laravel-livewire-tables#1653] [Q&A]
im using 10.000 dummy data, when use the select all button it took a minute to perform, and sometimes it freezes, is there another way or trick to resolve this?

**Comment by lrljoe** [comment]
So the reason it takes a little while, is it's having to retrieve all of the item IDs behind the scenes, based on your filters, searches etc, so that "Select All", reflects exactly what you've searched/filtered for.

The idea behind this is that you can "Select All", then navigate through the pages, and de-select certain items, prior to performing your Bulk Action.  This is more useful when you're "Activating" 998/1000 items for example.

It won't be in the next release, but what I can look to do is add in a toggle for the "Select All" behaviour.  My thinking is:
By Default: It'll either behave as current 
New Feature: It will do nothing in the frontend when you first click "Select All", and I'll add in a method for your Bulk Actions methods to retrieve the "All" items list.

What're your thoughts on the above?

**Reply by samods2** [reply]
@lrljoe Hello, I am also having the same issue but for me the screen freezes all the time when I check the select all checkbox. I am dealing with records more than 75000. I agree with what you have mentioned above adding a toggle for the "Select All" behavior. I hope you will be able to work on this as a higher priority task and provide the changes in the next release or asap.

Thank you

**Reply by lrljoe** [reply]
It's likely to make it in sometime in the middle of next month, as I'm several thousand miles from my computer at the moment and will be for another couple of weeks

**Reply by samods2** [reply]
@lrljoe Just following up on the same.

**Reply by samods2** [reply]
@lrljoe  Any update on this one?

**Reply by lrljoe** [reply]
Thanks for the reminder, I've added it into the backlog, there are a couple of tweaks ahead of it, but definitely will be in a release this month

**Reply by samods2** [reply]
@lrljoe Were you able to check this one?

**Reply by lrljoe** [reply]
I've looked at it briefly, and have tinkered around a bit with behaviours.

Adding in a config tweak for not immediately selecting all in the backend when clicking "Select All" is in the works, just trying to ensure that default behaviour remains.

**Reply by lrljoe** [reply]
So I've managed to produce a fix for this which involves using a new config option.  

The config option effectively disconnects the "selected" entanglement, so when using it you cannot "select all" and then "deselect some"
You can however still "select some".

I'll put together the relevant docs/tests over the next week

**Reply by lrljoe** [reply]
#1810 - This PR will address this issue, but I need to write some tests for it still.
